sight_service/message_queue.py

```diff
@@ -0,0 +1,400 @@
+"""A message queue implementation using reader-writer locks."""
+
+import abc
+import copy
+import enum
+from typing import Any, Callable, Dict, Generic, Optional, Protocol, TypeVar
+import uuid
+
+from helpers.logs.logs_handler import logger as logging
+from overrides import overrides
+from readerwriterlock import rwlock
+
+# Alias for message ID type
+ID = int
+
+
+class MessageLocation(enum.Enum):
```

Comment:
Small nit. MessageState is a clear name. Location implies something more like where it is stored or who received it.

Reply:
yes , this is a better name

Reply:
updated